src/caseless.rs

```diff
@@ -0,0 +1,134 @@
+//! This module contains a caseless filesystem.
+
+use std::ffi::OsString;
+use std::io;
+use std::path::{Component, Path, PathBuf};
+
+use crate::index::normalize_path;
+use crate::prelude::*;
+use crate::store::Entries;
+
+/// Caseless filesystem wrapping an inner filesystem.
+#[derive(Clone, Debug)]
+pub struct CaselessFs<T> {
+    /// Inner filesystem store.
+    inner: T,
+}
+
+impl<T: Store> CaselessFs<T> {
+    /// Creates a new caseless filesystem with the provided inner filesystem.
+    pub fn new(inner: T) -> Self {
+        Self { inner }
+    }
+
+    /// Moves the inner filesystem out of the caseless filesystem.
+    /// Inspired by std::io::Cursor.
+    pub fn into_inner(self) -> T {
+        self.inner
+    }
+
+    /// Gets a reference to the inner filesystem.
+    /// Inspired by std::io::Cursor.
+    pub fn get_ref(&self) -> &T {
+        &self.inner
+    }
+
+    /// Gets a mutable reference to the inner filesystem.
+    /// Inspired by std::io::Cursor.
+    pub fn get_mut(&mut self) -> &mut T {
+        &mut self.inner
+    }
+
+    /// Finds paths that match the caseless path.
+    pub fn find<P: AsRef<Path>>(&self, path: P) -> Vec<PathBuf> {
+        self.find_path(&normalize_path(path.as_ref()))
+    }
+
+    /// Finds paths that match the caseless path.
+    pub fn find_path(&self, path: &Path) -> Vec<PathBuf> {
```

Comment:
Can you implement this inside of `find<P>`, since `find_path` can be generated at compile time from the generic?

Reply:
Ok.
I copied the pattern of open/open_path and entries/entries_path.
I don't know why you have them so I just followed along.

Reply:
MiniFs has a linked list of trait objects (Store) which I can't call the generic methods `open` and `entries` on. Those are part of the autotrait StoreExt (not sure if it's the best way to do things though).